### PR TITLE
Fix soft cap in JAX Splash attention

### PR DESCRIPTION
**Pull request by huytransformer**
# Description

Fixes attention logit soft cap in the JAX Splash attention path. Example, gemma2 configs set `attn_logits_soft_cap`, but the `use_jax_splash` path did not pass that value into `flash_attention_block_masked`. 

# Tests

Ran:

```bash
python3 -m py_compile \
  src/maxtext/layers/attention_op.py \
  src/maxtext/kernels/attention/jax_flash_attention.py \
  tests/unit/attention_test.py

python3 -m pytest tests/unit/attention_test.py::JaxFlashAttentionTest::test_flash_attention_block_masked_soft_cap
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: tests/unit/kernels_test.py [hunk 1]
```diff
@@ -18,11 +18,47 @@
 
 import jax
 import jax.numpy as jnp
+from maxtext.kernels.attention import jax_flash_attention
 from maxtext.kernels.attention.ragged_attention import ragged_gqa, ragged_mha, ragged_mqa, reference_gqa, reference_mha, reference_mqa
 import numpy as np
 import pytest
 
 
+class JaxFlashAttentionTest(unittest.TestCase):
```
Comment (NuojCheng) [inline tests/unit/kernels_test.py]: jax flash attention is not kernel? maybe move to attention test

**Review by NuojCheng — APPROVED** [review]

**Review by igorts-git — APPROVED** [review]

File: tests/unit/kernels_test.py (hunk 1 above)
Comment (huytransformer) [reply tests/unit/kernels_test.py]: Done (for strange and weird reasons it was in kernels/ so I wasnt sure where to put the test)
